packages/neo-one-client-core/src/user/keystore/LedgerHandler.ts

```
@@ -0,0 +1,134 @@
+
+    return crypto.toECPoint(response).toString('hex');
+  }
+
+  public async sign({ message, account }: { readonly message: Buffer; readonly account: number }): Promise<Buffer> {
+    const data = Buffer.concat([message, BIP44(account)]);
+    const chunks = chunkBuffer(data, 250);
+    const initChunks = chunks.slice(0, chunks.length - 1);
+    const finalChunk = chunks[chunks.length - 1];
+
+    await Promise.all(
+      initChunks.map(async (chunk) => {
+        await this.send(Buffer.concat([initMessageParams, Buffer.from([chunk.length]), chunk]));
+      }),
+    );
+
+    return this.send(
+      Buffer.concat([finalMessageParams, Buffer.from(finalChunk.length.toString(16), 'hex'), finalChunk]),
+    );
+  }
+
+  public async close(): Promise<void> {
+    await this.ledger.close();
+  }
+
+  private async send(msg: Buffer): Promise<Buffer> {
+    // tslint:disable-next-line:promise-must-complete
+    const promise = new Promise<Buffer>((resolve, reject) => {
+      this.mutableQueue.push({
+        msg,
+        resolve,
+        reject,
+      });
+    });
+
+    this.startSendLoop().catch(() => {
+      // do nothing
+    });
+
+    return promise;
+  }
+
+  private async startSendLoop(): Promise<void> {
+    if (this.mutableRunning) {
+      return;
+    }
+
+    this.mutableRunning = true;
+
+    let entry = this.mutableQueue.shift();
+    // tslint:disable-next-line:no-loop-statement
+    while (entry !== undefined) {
+      try {
+        const result = await this.doSend(entry.msg);
+        entry.resolve(result);
+      } catch (error) {
+        entry.reject(error);
+      }
+      entry = this.mutableQueue.shift();
+    }
+
+    this.mutableRunning = false;
+  }
+
+  private async doSend(msg: Buffer): Promise<Buffer> {
+    const response = await this.ledger.exchange(msg);
+
+    const status = response.slice(response.length - 2);
+    if (status.toString('hex') !== '9000') {
+      throw new LedgerStatusCodeError(status.toString('hex'));
```

Let's throw a more specific error for the codes that we know about. For a code that we don't know, we can throw this general error.

---

Since this is the only issue, I'm going to go ahead and merge, but please make a followup PR to improve the errors here.

---

#524